### PR TITLE
Add new settings for adjusting the hit error display

### PR DESCRIPTION
**Pull request by peppy**
Based on feedback received since the last release (with the minor design update for this component). Also the first actual settings attached to a skin component, which will give people something to play with in the skin editor until more are added.

The animation of the component on initial display is also marginally better now.

https://user-images.githubusercontent.com/191335/158967930-5f0ce6f7-0e5b-4ca0-8b57-05c0468fb062.mp4

I can split this out into multiple pull requests if required (each setting is isolated to a commit, only kept in one PR to minimise conflict resolution).

### CODE REVIEW

File: osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs [hunk 1]
```diff
@@ -155,17 +155,187 @@ protected override void LoadComplete()
...
+                        Y = 10,
+                        Size = new Vector2(icon_size),
+                        Icon = FontAwesome.Solid.Bicycle,
+                        Anchor = Anchor.BottomCentre,
+                        Origin = Anchor.Centre,
+                    };
+
+                    break;
+
+                case LabelStyles.Text:
```
Comment (smoogipoo) [inline osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs:294]: It's bugging me that these text components aren't centred.

image

image

**Review by smoogipoo — CHANGES_REQUESTED** [review]
Also not related but:
```
[runtime] 2022-03-18 09:55:03 [error]: Skin component osu.Game.Screens.Play.HUD.HitErrorMeters.HitErrorMeter could not be loaded in the editor component list due to an error
[runtime] 2022-03-18 09:55:03 [error]: System.MissingMethodException: Cannot dynamically create an instance of type 'osu.Game.Screens.Play.HUD.HitErrorMeters.HitErrorMeter'. Reason: Cannot create an abstract class.
[runtime] 2022-03-18 09:55:03 [error]: at System.RuntimeType.ActivatorCache..ctor(RuntimeType rt)
[runtime] 2022-03-18 09:55:03 [error]: at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
[runtime] 2022-03-18 09:55:03 [error]: at osu.Game.Skinning.Editor.SkinComponentToolbox.attemptAddComponent(Type type) in /home/smgi/Repos/osu/osu.Game/Skinning/Editor/SkinComponentToolbox.cs:line 67
```

File: osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs:294]: The text version specifically? I didn't find it too offensive but can add an offset.

**Comment by peppy** [comment]
> Also not related but:
> 
> ```
> [runtime] 2022-03-18 09:55:03 [error]: Skin component osu.Game.Screens.Play.HUD.HitErrorMeters.HitErrorMeter could not be loaded in the editor component list due to an error
> [runtime] 2022-03-18 09:55:03 [error]: System.MissingMethodException: Cannot dynamically create an instance of type 'osu.Game.Screens.Play.HUD.HitErrorMeters.HitErrorMeter'. Reason: Cannot create an abstract class.
> [runtime] 2022-03-18 09:55:03 [error]: at System.RuntimeType.ActivatorCache..ctor(RuntimeType rt)
> [runtime] 2022-03-18 09:55:03 [error]: at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
> [runtime] 2022-03-18 09:55:03 [error]: at osu.Game.Skinning.Editor.SkinComponentToolbox.attemptAddComponent(Type type) in /home/smgi/Repos/osu/osu.Game/Skinning/Editor/SkinComponentToolbox.cs:line 67
> ```

Am aware of this, will fix separately.

File: osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs:294]: Please do, this stuck out to me like a sore thumb.

Comment (peppy) [reply osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs:294]: osu! 2022-03-18 at 11 17 17

Should be good now.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**